pkg/cri/resource-manager/policy/builtin/static-pools/config.go

```diff
@@ -81,17 +83,17 @@ var (
 	cpusetValidationRe = regexp.MustCompile(`^(([\d]+)|([\d]+-[\d]+))(,(([\d]+)|([\d]+-[\d]+)))*$`)
 )
 
-func parseConfData(raw []byte) (*conf, error) {
+func parseConfData(raw []byte) (pools, error) {
 	conf := &conf{}
 
 	err := yaml.Unmarshal(raw, &conf)
 	if err != nil {
 		return nil, stpError("Failed to parse config file: %v", err)
 	}
-	return conf, nil
+	return conf.Pools, nil
```

Comment:
With this implementation in place, isn't the behavior now such that `parseConfFile()` silently ignores extra configuration data which is present in the file but should only be given via global configuration (ConfigMap, fallback or forced configuration file) ?

If it is, wouldn't this be easily fixed by changing `parseConfData()` to internally use `pools{}` for unmarshalling instead of a full `&conf{}` ?

Reply:
Json/yaml parsing always silently ignores extra keys. But yes, I was thinking about this, too, but left it like this for backwards compatibility. I don't know if anyone is really using this so maybe we should just change it as you described. WDYT?

Reply:
What I meant is that now you explicitly parse the data as `&conf{}` but then only take the `.Pool` subset from it, explicitly ignoring yourself silently anything else that might have been given by the user. This is a bit different than the user feeding non-existent fields which are then ignored by the YAML unmarshaller. My suggestion was to instead explicitly unmarshal the data as `pools{}` instead. I think that would be structurally different enough from a `conf{}`, that data for the latter would hopefully not pass the YAML unmarshaller as an instance of the former.

Reply:
Yeah, that's true, I was cutting some corners there. I changed it now.

Reply:
Hmm, then I don't understand something. It still looks to me as though `parseConfData()` would unmarshal into a full `&conf{}`, which AFAICT allows the user to put a full configuration in the CMK-specific config file, with all other parts than conf.Pools getting silently ignored.